mainServiceController/src/main/java/org/venice/piazza/servicecontroller/async/AsynchronousServiceWorker.java

```diff
@@ -0,0 +1,342 @@
+import org.joda.time.DateTime;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.venice.piazza.servicecontroller.data.mongodb.accessors.MongoAccessor;
+import org.venice.piazza.servicecontroller.messaging.handlers.ExecuteServiceHandler;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import messaging.job.JobMessageFactory;
+import messaging.job.KafkaClientFactory;
+import model.job.result.type.DataResult;
+import model.job.result.type.ErrorResult;
+import model.job.type.ExecuteServiceJob;
+import model.response.JobResponse;
+import model.service.metadata.Service;
+import model.status.StatusUpdate;
+import util.PiazzaLogger;
+import util.UUIDFactory;
+
+/**
+ * This Worker will make the direct REST requests to that User Service, for execution, cancelling, and updating status.
+ * 
+ * @author Patrick.Doody
+ *
+ */
+@Component
+public class AsynchronousServiceWorker {
+	@Value("${async.status.endpoint}")
+	private String STATUS_ENDPOINT;
+	@Value("${async.results.endpoint}")
+	private String RESULTS_ENDPOINT;
+	@Value("${async.delete.endpoint}")
+	private String DELETE_ENDPOINT;
+	@Value("${vcap.services.pz-kafka.credentials.host}")
+	private String KAFKA_HOST_PORT;
+	@Value("${SPACE}")
+	private String SPACE;
+	@Value("${async.status.error.limit}")
+	private int STATUS_ERROR_LIMIT;
+
+	@Autowired
+	private MongoAccessor accessor;
+	@Autowired
+	private PiazzaLogger logger;
+	@Autowired
+	private ExecuteServiceHandler executeServiceHandler;
+	@Autowired
+	private UUIDFactory uuidFactory;
+
+	private RestTemplate restTemplate = new RestTemplate();
+	private Producer<String, String> producer;
+	private ObjectMapper objectMapper = new ObjectMapper();
+
+	@PostConstruct
+	public void initialize() {
+		String KAFKA_HOST = KAFKA_HOST_PORT.split(":")[0];
+		String KAFKA_PORT = KAFKA_HOST_PORT.split(":")[1];
+		producer = KafkaClientFactory.getProducer(KAFKA_HOST, KAFKA_PORT);
+	}
+
+	/**
+	 * Executes the Piazza Job Type
+	 * 
+	 * @param jobType
+	 *            The Piazza Job Type, describing everything about the Service execution.
+	 */
+	@Async
+	public void executeService(ExecuteServiceJob job) {
+		// Log the Request
+		logger.log(String.format("Processing Asynchronous User Service with Job ID %s", job.getJobId()), PiazzaLogger.INFO);
+		// Handle the external HTTP execution to the Service
+		ResponseEntity<String> response = executeServiceHandler.handle(job);
+		if (response.getStatusCode().is2xxSuccessful() == false) {
```

Comment:
Consider using `!response.getStatusCode().is2xxSuccessful()` instead of an == to a boolean primitive.